src/VGriveClient.vala

```diff
@@ -228,26 +228,41 @@ namespace App {
                 return false;
             }
         }
-        private void check_deleted_files () {
+        
+         private void check_deleted_files () {
             // Mira els fitxers que hi ha a la llibreria
             // Si no existeixen en local o en remot, el treu de la llibreria i l'elimina de on encara hi sigui
             if (!this.is_syncing ()) return;
+            
             var it = this.library.map_iterator ();
             bool exist_local, exist_remote, must_delete = false;
-            DriveFile remote_file;
+            DriveFile[] remote_files;
             string remote_id, filename, aux, lpath;
             Array<string> to_delete = new Array<string> ();
+            
             for (var has_next = it.next (); has_next; has_next = it.next ()) {
-                if (!this.is_syncing ()) return;
                 // Check local exists
                 lpath = it.get_value();
                 aux = it.get_key();
                 filename = lpath.split("/")[lpath.split("/").length-1];
                 exist_local = this.local_file_exists(lpath);
 
                 // Check remote exists
-                remote_id = this.get_file_id(lpath);
-                exist_remote = remote_id != null && remote_id != "";
+                /*
+                EXPERIMENTAL REMOTE CHECKING
+                Note: could be incremented checking if the remote file is trased or not. If trashed, exist_remote should be false, otherwise should be true
+                */
+                remote_files = this.list_files(-1, lpath, -1);
```

Comment:
The method `list_files` returns all the files childs of `lpath`. Why do you use the list method? 
It's more efficient to ask only for the file we are checking with `get_file_id` instead of listing all the files.

Reply:
I put this to try understand better why this delete function is deleting files that exists in both remote and local.

I have 3 guesses:
1. Could be the remote checking that don't retrieve the files properly; or
2. Could be the local checking that is retrieving the filenames wrong; or
3. The escape function that is somehow messing up with the file names.

My thought was try to change the remote checking to make sure it's not it.

I know that is not efficient and I don't want to leave like that, but I want to make sure that the bug keep occurring, so we can put the remote check aside and turn back into `get_file_id`

Makes sense what I'm trying to do?

Reply:
Yes, it make sense.

Where I have got more problems is in the escape function with filenames that have "special" characters. Could you give me some "problematic" filename that you have detected?

Reply:
Some special caracters were something like:
~
# 
accents in general (á ã õ ó)

But I don't think we need to escape this anymore. I made a lot of changes in the checking functions and seems to work fine with caracters.